yaksh/views.py

```diff
@@ -3449,3 +3490,93 @@ def hide_comment(request, course_id, uuid):
+    if not course.is_creator(user) and not course.is_teacher(user):
+        raise Http404('This course does not belong to you')
+
+    quiz = get_object_or_404(Quiz, pk=quiz_id)
+    student = get_object_or_404(User, pk=user_id)
+
+    if not course.is_enrolled(student):
+        raise Http404('The student is not enrolled for this course')
+
+    micromanager, created = MicroManager.objects.get_or_create(
+        course=course, student=student, quiz=quiz
+    )
+    micromanager.manager = user
+    micromanager.save()
+
+    if (not micromanager.is_special_attempt_required() or
+            micromanager.is_last_attempt_inprogress()):
+        name = student.get_full_name()
+        msg = '{} can attempt normally. No special attempt required!'.format(
+            name)
+    elif micromanager.can_student_attempt():
+        msg = '{} already has a special attempt!'.format(
+            student.get_full_name())
+    else:
+        micromanager.allow_special_attempt()
+        msg = 'A special attempt is provided to {}!'.format(
+            student.get_full_name())
+
+    messages.info(request, msg)
+    return my_redirect('/exam/manage/monitor/{0}/{1}/'.format(quiz_id,
+                                                              course_id))
+
+
+@login_required
+@email_verified
+def revoke_special_attempt(request, micromanager_id):
+    user = request.user
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page')
+
+    micromanager = get_object_or_404(MicroManager, pk=micromanager_id)
+    course = micromanager.course
+    if not course.is_creator(user) and not course.is_teacher(user):
+        raise Http404('This course does not belong to you')
+    micromanager.revoke_special_attempt()
+    msg = 'Revoked special attempt for {}'.format(
+        micromanager.student.get_full_name())
+    messages.info(request, msg)
+    return my_redirect('/exam/manage/monitor/{0}/{1}/'.format(
+        micromanager.quiz.id, course.id))
+
+
+@login_required
+@email_verified
+def extend_time(request, paper_id):
+    user = request.user
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page')
+
+    anspaper = get_object_or_404(AnswerPaper, pk=paper_id)
+    course = anspaper.course
+    if not course.is_creator(user) and not course.is_teacher(user):
+        raise Http404('This course does not belong to you')
+
+    if request.method == "POST":
+        extra_time = request.POST.get('extra_time', None)
+        if extra_time is None:
+            msg = 'Please provide time'
+        else:
+            anspaper.set_extra_time(extra_time)
+            msg = 'Extra {0} minutes given to {1}'.format(
+                extra_time, anspaper.user.get_full_name())
+    else:
+        msg = 'Bad Request'
+    messages.info(request, msg)
+    return my_redirect('/exam/manage/monitor/{0}/{1}/'.format(
+        anspaper.question_paper.quiz.id, course.id))
```

Comment:
Change all the hardcoded url to reverse url django format.
You can change in this PR or the follow up PR

Reply:
This can be done in the follow up PR